### PR TITLE
CUDA 11.1.0

### PR DESCRIPTION
**Pull request by ax3l**
Add the latest CUDA release (11.1.0) and update its compiler conflicts.

https://gist.github.com/ax3l/9489132

cc @svenevs 

- [x] rebase after #19035 was merged.

### CODE REVIEW

File: lib/spack/spack/build_systems/cuda.py [hunk 1]
```diff
@@ -129,7 +133,9 @@ def cuda_flags(arch_list):
     conflicts('%gcc@8:', when='+cuda ^cuda@:10.0.130' + arch_platform)
     conflicts('%gcc@9:', when='+cuda ^cuda@:10.1.243' + arch_platform)
     # officially, CUDA 11.0.2 only supports the system GCC 8.3 on ppc64le
-    conflicts('%gcc@:4,10:', when='+cuda ^cuda@:11.0.2' + arch_platform)
+    conflicts('%gcc@:4', when='+cuda ^cuda@11.0.2' + arch_platform)
```
Comment (ax3l) [inline lib/spack/spack/build_systems/cuda.py]: Fixed in #19035:
```suggestion
    conflicts('%gcc@:4', when='+cuda ^cuda@11.0.2:' + arch_platform)
```

**Review by sethrj — APPROVED** [review]
Looks good to me.

File: lib/spack/spack/build_systems/cuda.py [hunk 2]
```diff
@@ -95,14 +95,16 @@ def cuda_flags(arch_list):
...
+    conflicts('%gcc@11:', when='+cuda ^cuda@:11.1.0' + arch_platform)
     conflicts('%pgi@:14.8', when='+cuda ^cuda@:7.0.27' + arch_platform)
     conflicts('%pgi@:15.3,15.5:', when='+cuda ^cuda@7.5' + arch_platform)
     conflicts('%pgi@:16.2,16.0:16.3', when='+cuda ^cuda@8' + arch_platform)
     conflicts('%pgi@:15,18:', when='+cuda ^cuda@9.0:9.1' + arch_platform)
     conflicts('%pgi@:16,19:', when='+cuda ^cuda@9.2.88:10' + arch_platform)
     conflicts('%pgi@:17,20:',
               when='+cuda ^cuda@10.1.105:10.2.89' + arch_platform)
-    conflicts('%pgi@:17,20.2:', when='+cuda ^cuda@11.0.2' + arch_platform)
+    conflicts('%pgi@:17,20.2:',
```
Comment (ax3l) [inline lib/spack/spack/build_systems/cuda.py]: According to the headers, this is all that is checked (don't trust the compiler table, bug reports opened)
```suggestion
    conflicts('%pgi@:17,21:',
```

**Review by svenevs — APPROVED** [review]
Awesome, thanks!  (And for getting the right results not following table xD)

**Review by alalazo — APPROVED** [review]

**Pull request merged**